tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -0,0 +1,347 @@
+
+    assert query_equalize(statement) == query_equalize(f"""
+            SELECT
+                *
+            FROM
+                {sf_adapter.quoted_dot_notation(relation)}
+            """)
+
+    assert len(sf_adapter._safe_query(statement)) > 0
+
+
+def test_union_constraint_statement(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "POLYMORPHIC_DATA", "PARENT_TABLE_2"
+    subject = Relation(database=DATABASE,
+                       schema=SCHEMA,
+                       name=TABLE,
+                       materialization=TABLE,
+                       attributes=[])
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "POLYMORPHIC_DATA", "CHILD_TYPE_1_ITEMS"
+    constraint = Relation(database=DATABASE,
+                          schema=SCHEMA,
+                          name=TABLE,
+                          materialization=TABLE,
+                          attributes=[])
+    max_number_of_outliers = randrange(10)
+    subject_key, constraint_key = "ID", "PARENT_2_ID"
+    statement = sf_adapter.union_constraint_statement(subject, constraint, subject_key, constraint_key,
+                                                      max_number_of_outliers)
+
+    assert query_equalize(statement) == query_equalize(f"""
+            (SELECT
+                *
+            FROM
+            {sf_adapter.quoted_dot_notation(subject)}
+            WHERE
+                {subject_key}
+            NOT IN
+            (SELECT
+                {constraint_key}
+            FROM
+            {sf_adapter.quoted_dot_notation(constraint)})
+            LIMIT {max_number_of_outliers})
+            """)
+
+    assert len(sf_adapter._safe_query(statement)) > 0
+
+
+def test_polymorphic_constraint_statement(sf_adapter):
+    DATABASE, SCHEMA, TABLE = "SNOWSHU_DEVELOPMENT", "POLYMORPHIC_DATA", "PARENT_TABLE"
+    LOCAL_KEY, REMOTE_KEY = "ID", "CHILD_ID"
+    LOCAL_TYPE, TYPE_MATCH_VAL = "CHILD_TYPE", "type_2"
+    relation = Relation(database=DATABASE,
+                        schema=SCHEMA,
+                        name=TABLE,
+                        materialization=TABLE,
+                        attributes=[])
+    relation.core_query = f"""
+        SELECT
+            *
+        FROM 
+            {DATABASE}.{SCHEMA}.{TABLE}
+            SAMPLE BERNOULLI (10)
+        """
+    predicate = sf_adapter.predicate_constraint_statement(relation, True, LOCAL_KEY, REMOTE_KEY)
+
+    assert f" ({predicate} AND LOWER({LOCAL_TYPE}) = LOWER('{TYPE_MATCH_VAL}') ) " == \
+           sf_adapter.polymorphic_constraint_statement(relation,
+                                                       True,
+                                                       LOCAL_KEY,
+                                                       REMOTE_KEY,
+                                                       LOCAL_TYPE,
+                                                       TYPE_MATCH_VAL)
+
+
+def test_build_conn_string(sf_adapter):
```

Comment:
This one could probably be moved to the unit tests file as well or merged with the existing `test_conn_string_basic` test there by adding a randomly generated `ROLE` variable.